code/modules/vehicles/mecha/equipment/tools/mining_tools.dm

```diff
@@ -25,24 +25,33 @@
 	AddComponent(/datum/component/butchering, 50, 100, null, null, TRUE)
 
 /obj/item/mecha_parts/mecha_equipment/drill/action(mob/source, atom/target, params)
+	// Check if we can even use the equipment to begin with.
 	if(!action_checks(target))
 		return
-	if(isspaceturf(target))
+
+	// We can only drill non-space turfs, living mobs and objects.
+	if(isspaceturf(target) || !isliving(target) || !isobj(target) || !isturf(target))
```

Comment:
```suggestion
	if(isspaceturf(target) || !(isliving(target) || isobj(target) || isturf(target))
```

Reply:
Missed a bracket. Otherwise, if you drill and object, it'd return since it's not a living mob.